### ISSUE TITLE
macOS compatibility?

### ISSUE THREAD

**Issue by zachvalenti** [2birds/Fledermaus#5]
Thanks for making this project! I'm exploring using the Leap 2 as a pointer input with less risk for repetitive use injury as a podcast and video editor.

I managed to build and run this project on my 16" M1 Max laptop, but there was an immediate error reading the config file at the start. After successfully detecting my device, the app just idles, unresponsive to anything but "quit" with the message "Received unsupported message."

I wouldn't be surprised if I made a mistake somewhere (maybe I set up RapidJSON the wrong way), but I'm inexperienced enough that I don't know how to check. I'd appreciate any direction at all.

**Comment by 2birds** [comment]
Hi Zach,

Well done for getting it built on Mac. As it stands, it won't work with Mac as I haven't implemented the cursor control commands for OSX. I will do that at some point, but I can't give a timescale, unfortunately. The config file error is probably another OSX-specific issue, I don't think it would even build without RapidJSON being in the right place. I expect it's an easy fix, but again, I really haven't tried very hard to make it work for Mac.

TL;DR, it's not you, it's me needing to do more 🙃 

Stay tuned, and hopefully I'll have it working on Mac soon.

**Comment by zachvalenti** [comment]
Thank you! Tbh I wouldn't have gotten as far as I did without not one, but two LLM's suggesting next steps 😅

If I had more time I would keep going and attempt my first PR, but alas, I need to go back to the work I was hoping to be able to do by waving my hand around in the air 👋 Definitely will keep an eye here and give whatever version of Leap is available a try whenever it's ready for primetime.

Curious to know if the performance of your app is more usable than the included "Widgets" app "Pointer" demo? I found it too impercise for mousing around in REAPER (audio editing).

My sense is I'd probably see better results working out narrower, app-specific functions vs full system input control. REAPER specifically has a robust API/scripting system, but I'll need to wait until I have more free time to figure out plumbing between the sensor's output and *.lua scripts.

**Comment by 2birds** [comment]
I'm absolutely delighted to have confounded an LLM 😁  If you really want to see them struggle, ask them to do cryptic crosswords.

I'm a former Ultraleap employee, and I created Fledermaus because Pointer didn't do quite what I wanted it to 😂 The Widgets were intended to be very lightweight, and allow you to compose a useful setup by using a subset of them. Fledermaus performs well on SIR170 and LMC2, not so much on the original LMC.

Interesting that you want it for REAPER. I use REAPER too, and built a VST3 plugin that uses Ultraleap hand tracking..

The problem you might have scripting a plugin for REAPER is that its scripting language is Lua. Ultraleap doesn't have a plugin for Lua, so you'd have to build that first. If you can get your hands on ultraleap-tracking-websocket (an unofficial revival of the old javascript server) you might be able to open a websocket with Lua and do something with the data being chucked over the LeapJS port.

Good luck, and keep an eye on Fledermaus, I'll sort OSX support..eventually

**Comment by 2birds** [comment]
Hi Zach, I now have access to an Intel Mac and have implemented the missing cursor control.

There is a Fledermaus branch called "OSX", and this works on my machine! Let me know if you have any problems with it.

**Comment by zachvalenti** [comment]
Thank you so much for putting in the work to make it accessible!!!! (This is my first experience with a repo getting updated for asking, hence exclamation points)

I returned my original test unit so I could order a Svalboard to solve my immediate RSI issues while pulling overtime on a big sound design deadline. But I'll have to rebuy an LMC2 to try this out now!

Very cool that you're also a REAPER user and I am very intrigued by this VST3 plugin and how you use it! Is it available?

**Comment by 2birds** [comment]
You're welcome, I'm glad someone is interested in using it. Even more glad that you're trying to use it to help with RSI - that makes so much sense and I hadn't thought of it before. I love that it is not being used for evil 😆 

I haven't put the VST3 up anywhere yet. It was (intentionally) basic, but did the trick! It'll end up on this github account at some point. Like and subscribe, etc.
